test/JDBC/expected/TestSpatialFun-vu-verify.out

```diff
@@ -0,0 +1,932 @@
+0
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+DECLARE @equal BIT = 1;
+SELECT ROUND(PointColumn.STEquals(@referencePoint) / @equal, 0) * @equal AS Equalitygroup,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STEquals(@referencePoint) / @equal, 0) * @equal
+ORDER BY Equalitygroup;
+GO
+~~START~~
+numeric#!#int
+1#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(1.0, 0.0, 4326);
+SELECT ROUND(PointColumn.STEquals(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX AS Equalitygroup,
+COUNT(*) AS PointCount
+FROM YourTableTemp
+GROUP BY ROUND(PointColumn.STEquals(@referencePoint) / @referencePoint.STX, 0) * @referencePoint.STX
+ORDER BY Equalitygroup;
+GO
+~~START~~
+float#!#int
+0.0#!#2
+~~END~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn1.STEquals(PointColumn2) AS equal,
+cast(PointColumn1.STEquals(@referencePoint) as float) - LAG(PointColumn1.STEquals(PointColumn2)) OVER (ORDER BY ID) AS Equalitygroup 
+FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
+SELECT ID, PointColumn.STEquals(@referencePoint) AS equal,
+cast(@referencePoint.STContains(PointColumn) as float) - LAG(@referencePoint.STX) OVER (ORDER BY ID) AS Equalitygroup
+FROM YourTableTemp ORDER BY PointColumn.STX;
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: cannot cast type "bit" to double precision)~~
+
+
+
+
+
+
+--For STContains
+DECLARE @point1 geometry, @point2 geometry;
+SET @point1 = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SET @point2 = geometry::STGeomFromText('POINT(-122.35000 47.65000)', 4326);
+SELECT STContains(@point1, @point2);
+go
+~~START~~
+bit
+0
+~~END~~
+
+
+DECLARE @point1 geometry, @point2 geometry;
+SET @point1 = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SET @point2 = geometry::STGeomFromText('POINT(-122.35000 47.65000)', 4326);
+SELECT @point1.STContains(@point2);
+go
+~~START~~
+bit
+0
+~~END~~
+
+
+DECLARE @point1 geometry, @point2 geometry;
+SET @point1 = geometry::STPointFromText('POINT(-122.34900 47.65100)', 4326);
+SET @point2 = geometry::STGeomFromText('POINT(-122.35000 47.65000)', 4326);
+SELECT @point1 . STContains ( @point2 );
+Go
+~~START~~
+bit
+0
+~~END~~
+
+
+SELECT ID, PointColumn1.STContains(PointColumn2) AS contain FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
+~~START~~
+int#!#bit
+1#!#0
+~~END~~
+
+
+DECLARE @point1 geometry = geometry::Point(1.0, 2.0, 4326);
+SELECT ID, PointColumn1.STContains(@point1) AS contain FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
+~~START~~
+int#!#bit
+1#!#0
+~~END~~
+
+
+DECLARE @point1 geometry = geometry::Point(1.0, 2.0, 4326);
+SELECT ID, @point1.STContains(PointColumn2) AS contain FROM YourTableTemp2 ORDER BY PointColumn1.STX;
+GO
```

Comment:
What happens when we do geography.Contains/equals(geometry) or vice-versa? Let's have such test case as well.

Reply:
Resolved added these test cases. It will give error not allow to perform this operation.